From 42892117783e2fffe8a36103e1ff3f770edcb838 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Mon, 11 Jun 2018 17:50:19 +0530
Subject: [PATCH 0364/1852] drm: xlnx: vtc: Return IRQ_NONE for no interrupt

commit f7c7e01f86a120976267bf72b840b17a316b2881 from
https://github.com/Xilinx/linux-xlnx.git

Return IRQ_NONE in the interrupt handler, when there's no interrupt.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_vtc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_vtc.c b/drivers/gpu/drm/xlnx/xlnx_vtc.c
index aa04f7757174..019c0f7a5455 100644
--- a/drivers/gpu/drm/xlnx/xlnx_vtc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_vtc.c
@@ -509,6 +509,9 @@ static irqreturn_t xilinx_vtc_intr_handler(int irq, void *data)
 
 	u32 intr = xlnx_vtc_intr_get(vtc);
 
+	if (!intr)
+		return IRQ_NONE;
+
 	if ((intr & VTC_IXR_G_VBLANK) && (vtc->vblank_fn))
 		vtc->vblank_fn(vtc->vblank_data);
 
-- 
2.31.1

